paraatm/bdn/_bdn.py

```diff
@@ -0,0 +1,228 @@
+        https://en.wikipedia.org/wiki/Deep_learning#Deep_neural_networks
+        
+        Returns
+        -------
+        A sequential dnn model ready to be fit in 
+        """
+        if len(self.x_train.shape)!=2:
+            raise Exception('expect a 2 dimension array!')
+        in_shape = self.x_train.shape[-1]
+        model=Sequential()
+        model.add(Dense(self.num_unit,kernel_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay),
+                        kernel_initializer='random_uniform',
+                        input_dim=in_shape))
+        
+        model.add(Activation('relu'))
+        model.add(Dropout(self.idrop))
+        model.add(Dense(self.num_unit,kernel_regularizer=l2(self.weight_decay),
+                       bias_regularizer=l2(self.weight_decay),
+                        kernel_initializer='random_uniform',
+                       ))
+        model.add(Dropout(self.idrop))
+        model.add(Activation('relu'))  
+        if self.odrop:
+            model.add(Dropout(self.odrop))
+        model.add(Dense(1,activation='linear',
+                        kernel_regularizer=l2(self.weight_decay),
+                        bias_regularizer=l2(self.weight_decay)))
+        optimizer_=Adam(self.lr)
+        model.compile(loss='mse',metrics=['mse'],optimizer=optimizer_)
+        return model        
+        
+
+    def pred(self, x_test, y_test, get_var = True,iter_=5):
+        """
+        pred is used to generate results for the deep learning model. To get uncertainty of the model, get_var should be set as 'True'. By doing this, multiple realizations are done for one test data. The output of the pred method is the raw prediction results. Below is a simple example of how to build a bayesian RNN model.
+        
+        from paraatm.bdn import Bdn
+        import numpy as np
+        rnn_model = Bdn(RNN_x_train,RNN_y_train)
+        rnn_y_test_pred = rnn_model.pred(RNN_x_test, RNN_y_test)
+        ### to obtain the mean and variance of the prediction:
+        mean_pred = np.mean(rnn_y_test_pred,axis=0)
+        var_pred = np.var(rnn_y_test_pred,axis=0)
+        
+        
+        Parameters
+        ----------
+        x_test: array
+            test data   
+        y_test: array
+            test target   
+        get_var: Boolen
+            Whether the prediction result is deterministic or with uncertainty
+            By default: it's set as 'True'
+            get_var == True: prediction with uncertainty
+            get_var == False: deterministic result  
+        iter_: int
+            Number of realizations for each sample
+            By default, it's set as 5
+            This is only used when get_var == True
+        
+        Returns
+        -------
+        An array which is prediction for the test data
+        The array has a shape as num_iter by num_test_data
+        num_test_data: number of test data
+        num_iter: number of iterations
```

Comment:
I still do not think this is a complete description of the return value.

1. When I call this function with `get_var=False`, the shape of the results is not as described in these comments.
2. The first line states that the return value is "an array which is prediction for the test data".  This applies only to the case when `get_var=False`, right?  If `get_var=True`, do I get variance values instead of predicted values (means)?